synapse/federation/federation_server.py

```diff
@@ -541,11 +577,11 @@ def on_query_client_keys(self, origin, content):
         return {"events": [ev.get_pdu_json(time_now) for ev in missing_events]}
 
     @log_function
-    def on_openid_userinfo(self, token):
+    def on_openid_userinfo(self, token: str) -> Optional[str]:
```

Comment:
this looks like it's going to return a deferred... maybe convert it to async/await?

Reply:
That's simpler!